vmx/tests/counter.rs

```diff
@@ -56,11 +56,52 @@ fn counter_read_write_session() -> Result<(), Error> {
 
     other_session.transact::<(), ()>(id, "increment", ())?;
 
-    let _commit_id = other_session.commit(&id)?;
+    let _commit_id = vm.commit(&id)?;
 
     // session committed, new value accessible
 
     assert_eq!(vm.query::<(), i64>(id, "read_value", ())?, 0xfd);
 
     Ok(())
 }
+
+#[test]
+fn counter_commit_restore() -> Result<(), Error> {
+    let mut vm = VM::ephemeral()?;
+    let id = vm.deploy(module_bytecode!("counter"))?;
+
+    // commit 1
+    let mut session_1 = vm.session();
+
+    assert_eq!(session_1.query::<(), i64>(id, "read_value", ())?, 0xfc);
+
+    session_1.transact::<(), ()>(id, "increment", ())?;
+
+    let commit_1 = vm.commit(&id)?;
```

Comment:
I don't think commits should be at a module level, they should rather commit the result of a session. The user should have no control over which modules are persisted at all.